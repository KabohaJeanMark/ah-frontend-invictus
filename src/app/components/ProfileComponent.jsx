import React from 'react';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faEdit } from '@fortawesome/free-solid-svg-icons';
import '../../assets/styles/userprofile.scss';
import ProfileArticleComponent from './ProfileArticleComponent';
import NavbarComponent from '../components/Navbar';
import { SpinnerComponent } from './SpinnerComponent';


const ProfileComponent = (props) => {
  const { username, image } = props.state.profile.profile;
  const { isLoading } = props.state;
  return (
    <div className="user-profile-container">
      <div className="row">
        <div className="col-md-12"><NavbarComponent /></div></div>
      {isLoading === true ? (<SpinnerComponent />) : (<div className="row">
        <div className="col-md-1 " /><div className="col-md-5  thepro">
          <div className="image-div" >
            {image === "" ? (<img className="profile-image" alt="profileimage"
              src={require("../../assets/images/avatar2.png")} />) :
              (<img className="profile-image" alt="profileimage" src={image} />)}
          </div>
        </div>
        <div className="col-md-5 user-info">
          <div className="user-details">
            <h2 className="mb-4 username">{`${username}`}</h2>
            <p className=" mb-3 mt-4">
              <span className="mr-3 ml-5 font-weight-bold">4789</span>
              <span className="text-muted">Followers</span>
              <span className=" ml-5 mr-3 font-weight-bold">2134</span>
              <span className="text-muted">Following</span></p>
            <Link to="/editprofile" className="edit-button">
              <FontAwesomeIcon icon={faEdit} />Edit Profile</Link>
          </div>
        </div>
        <div className="col-md-1" />
        <ProfileArticleComponent /></div>)}
    </div>
  );
};

export default ProfileComponent;
