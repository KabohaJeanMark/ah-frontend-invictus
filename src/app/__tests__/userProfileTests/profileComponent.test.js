import React from 'react';
import { shallow } from 'enzyme';
import ProfileComponent from '../../components/ProfileComponent';

describe('<ProfileComponent />', () => {
  let wrapper;
  const props = {
    state: {
      profile: {
        profile: {
          firstname: 'edna',
          lastname: 'nakajugo',
          username: 'nakajugo',
          bio: 'i love singing'
        }

      }
    }
  };
  it('profile component renders without fail', () => {
    wrapper = shallow(<ProfileComponent {...props} />);
  });
  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
