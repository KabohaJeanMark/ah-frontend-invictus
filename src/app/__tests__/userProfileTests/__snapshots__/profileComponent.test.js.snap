// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileComponent /> matches snapshot 1`] = `ShallowWrapper {}`;
